crates/subspace-farmer/src/commands/farm.rs

```diff
@@ -34,18 +33,20 @@ pub async fn farm(base_directory: PathBuf, ws_server: &str) -> Result<()> {
     let identity = Identity::open_or_create(&base_directory)?;
 
     // start the farming task
-    // right now the instance is unused, however, if we want to call stop the process
-    // we can just drop the instance, and it will be stopped magically :)
-    let _farming_instance = Farming::start(
+    let farming_instance = Farming::start(
         plot.clone(),
         commitments.clone(),
         client.clone(),
         identity.clone(),
     );
 
     // start the background plotting
-    let _plotting_instance = Plotting::start(plot, commitments, object_mappings, client, identity);
+    let plotting_instance = Plotting::start(plot, commitments, object_mappings, client, identity);
 
-    Ok(()) // this is a placeholder at the moment, needs to wait on the instances in order to
-           // detect errors. Will be present in the next commit for modularity
+    tokio::select! {
+        _ = plotting_instance.wait() => {},
+        _ = farming_instance.wait() => {},
+    }
+
+    Ok(())
```

Comment:
This is a weird combo. If any of those return `Err()`, `Err()` should be returned from this function too.